src/ts/components/subtitleoverlay.ts

```diff
@@ -339,34 +365,52 @@ class ActiveSubtitleManager {
     return id;
   }
 
-  /**
-   * Adds a subtitle cue to the manager and returns the label that should be added to the subtitle overlay.
-   * @param event
-   * @return {SubtitleLabel}
-   */
-  cueEnter(event: SubtitleCueEvent): SubtitleLabel {
-    let id = ActiveSubtitleManager.calculateId(event);
+  cueEnter(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    this.addCueToMap(event, label);
+  }
 
-    let label = new SubtitleLabel({
-      // Prefer the HTML subtitle text if set, else try generating a image tag as string from the image attribute,
-      // else use the plain text
-      text: event.html || ActiveSubtitleManager.generateImageTagText(event.image) || event.text,
-      vtt: event.vtt,
-      region: event.region,
-      regionStyle: event.regionStyle,
-    });
+  cueUpdate(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    this.addCueToMap(event, label);
+  }
+
+  getPreviousCue(event: SubtitleCueEvent): SubtitleLabel {
+    return this.removeCueFromMap(event);
+  }
+
+  private addCueToMap(event: SubtitleCueEvent, label: SubtitleLabel): void {
+    let id = ActiveSubtitleManager.calculateId(event);
 
     // Create array for id if it does not exist
     this.activeSubtitleCueMap[id] = this.activeSubtitleCueMap[id] || [];
 
     // Add cue
     this.activeSubtitleCueMap[id].push({ event, label });
     this.activeSubtitleCueCount++;
+  }
 
-    return label;
+  private removeCueFromMap(event: SubtitleCueEvent): SubtitleLabel {
```

Comment:
As per code talk, we should use `undefined` instead of nulls
+ Missing return type for undefined
```suggestion
  private removeCueFromMap(event: SubtitleCueEvent): SubtitleLabel | undefined {
```

Reply:
Thanks for spotting, fixed in https://github.com/bitmovin/bitmovin-player-ui/pull/378/commits/6b9d35304a117fb79d93fe0fad303c7e013b989d